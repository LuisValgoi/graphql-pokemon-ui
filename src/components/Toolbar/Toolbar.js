import React from 'react';

import Dropdown from 'react-bootstrap/Dropdown';
import DropdownButton from 'react-bootstrap/DropdownButton';
import InputGroup from 'react-bootstrap/InputGroup';
import FormControl from 'react-bootstrap/FormControl';
import Container from 'react-bootstrap/Container';
import Col from 'react-bootstrap/Col';
import Row from 'react-bootstrap/Row';

const Toolbar = ({ amountSelected, onScrollAmountSelect }) => {
  const handleOnScrollAmountSelect = (e) => {
    const value = e.target.getAttribute('value');
    onScrollAmountSelect(parseInt(value, 10));
  };

  return (
    <Container>
      <Row className='pt-3 pb-3'>
        <Col sm={2}>
          <DropdownButton title={`Scroll Amount - (${amountSelected})`}>
            <Dropdown.Item value={1} onClick={(e) => handleOnScrollAmountSelect(e)}>1</Dropdown.Item>
            <Dropdown.Item value={2} onClick={(e) => handleOnScrollAmountSelect(e)}>2</Dropdown.Item>
            <Dropdown.Item value={3} onClick={(e) => handleOnScrollAmountSelect(e)}>3</Dropdown.Item>
          </DropdownButton>
        </Col>
        <Col sm={10} >
          <InputGroup >
            <FormControl placeholder="Search by a pokemon name" />
          </InputGroup>
        </Col>
      </Row>
    </Container>
  )
}

export default Toolbar;
