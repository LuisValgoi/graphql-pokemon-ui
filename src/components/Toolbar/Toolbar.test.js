import React from 'react'

import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Toolbar from './Toolbar';

afterEach(cleanup);

test('should render', () => {
  const { asFragment } = render(<Toolbar />);
  expect(asFragment()).toMatchSnapshot();
});

test('should have placeholder w/ the text: "Search by a pokemon name"', () => {
  const { getByTestId } = render(<Toolbar />);
  expect(getByTestId('toolbar-input')).toBeInTheDocument();
  expect(getByTestId('toolbar-input')).toHaveAttribute('placeholder', 'Search by a pokemon name');
});

test('should set variable through IoC to "text"', () => {
  let variable = '';
  const onSearch = (e) => variable = e;
  const utils = render(<Toolbar onSearch={onSearch} />);
  const input = utils.getByTestId('toolbar-input');
  fireEvent.input(input, { target: { value: 'text' } });
  expect(input.value).toBe(variable);
});
