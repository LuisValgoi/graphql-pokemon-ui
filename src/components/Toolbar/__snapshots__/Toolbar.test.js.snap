// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<DocumentFragment>
  <div
    class="row"
    style="margin-top: 1rem; padding-bottom: 1rem;"
  >
    <div
      class="col-sm-12"
    >
      <div
        class="input-group"
      >
        <input
          class="form-control"
          data-testid="toolbar-input"
          placeholder="Search by a pokemon name"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
